crates/runtime/src/accelerated_table.rs

```diff
@@ -91,6 +93,27 @@ pub enum Error {
     FailedToWriteData {
         source: datafusion::error::DataFusionError,
     },
+
+    #[snafu(display("The accelerated table does not support delete operations"))]
+    AcceleratedTableDoesntSupportDelete {},
+
+    #[snafu(display("Expected schema to have field '{field_name}' schema={schema}"))]
+    ExpectedSchemaToHaveField {
+        field_name: String,
+        schema: SchemaRef,
+    },
+
+    #[snafu(display("Expected field in schema '{field_name}' to have type '{expected_data_type}' schema={schema}"))]
```

Comment:
> `schema={schema}"`
Could make very large error messages for common tables (eg. tpch).

Reply:
This technically should never happen, so I wanted to provide as much debugging context as possible if it does.

Reply:
Good point